graphs/lca.cpp

```diff
@@ -1,35 +1,39 @@
+#include <bits/stdc++.h>
+using namespace std;
 struct lca {
+	using pii = pair<int, int>;
 	int Log;
 	vector<vector<int>> memo;
-	vector<int> depth;
+	vector<int> depth, dist;
 
-	lca(const vector<vector<int>> &graph, int root) {//0 - based nodes
+	// use weights of 1 for unweighted tree
+	lca(const vector<vector<pii>> &graph, int root = 0) {//0 - based nodes
 		int n = graph.size();
 		Log = 1;
 		while((1<<Log) < n) ++Log;
 		depth.resize(n,0);
+		dist.resize(n,0);
```

Comment:
while were here, can we initialize these in the initializer list - i.e. something like lca() : depth(n,0)